### DISCUSSION TITLE
Memory usage of repeated Headers

### DISCUSSION THREAD

**Discussion by wayneseguin** [stevehansen/csv#63] [Q&A]
When parsing CSV files using something like ReadFromStream or any variant, it creates a list of ICsvLine. Each ICsvLine has a Header object which is a string[]. Are these all references to a single string[]? Or are there "n" copies of the same header object, one per ICsvHeader. I'm loading some very large files and memory usage is massive. Wondering if it is just the data or if repeated headers are part of the problem. Thanks.

**Comment by BenjiAtTrystar** [comment]
I know this is old, but I figure I might as well provide a current answer for anyone who is looking.

In [CsvReader](https://github.com/stevehansen/csv/blob/master/Csv/CsvReader.cs#L417) you can see that the header array is generated once and then the same reference is passed to each ICsvLine.
